mmaction/models/recognizers/recognizer2d.py

```diff
@@ -57,13 +54,27 @@ def forward_test(self, imgs):
             losses.update(loss_aux)
             num_segs = 1
 
+        # When using `TSNHead` or `TPNHead`, shape is [batch_size, num_classes]
+        # When using `TSMHead`, shape is [batch_size * num_crops, num_classes]
+        # `num_crops` is calculated by:
+        #   1) `twice_sample` in `SampleFrames`
+        #   2) `num_sample_positions` in `DenseSampleFrames`
+        #   3) `ThreeCrop/TenCrop/MultiGroupCrop` in `test_pipeline`
+        #   4) `num_clips` in `SampleFrames` or its subclass if `clip_len != 1`
         cls_score = self.cls_head(x, num_segs)
-        if test_crops is not None:
-            if twice_sample:
-                test_crops = test_crops * 2
-            cls_score = self.average_clip(cls_score, test_crops)
 
-        return cls_score.cpu().numpy()
+        assert cls_score.size()[0] % batches == 0
+        if cls_score.size()[0] != batches:
```

Comment:
It seems that this `if` is unnecessary. Although `average_clip` has no effect when `cls_score.size()[0] == batches`, it can still generate cls_score after softmax when setting `prob`